x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -1361,4 +1434,23 @@ protected List<String> filteredWarnings() {
         return withDefaultLimitWarning(super.filteredWarnings());
     }
 
+    private static LogicalPlan analyzeWithEmptyFieldCapsResponse(String query) throws IOException {
+        IndexResolution resolution = IndexResolver.mergedMappings(
+            EsqlDataTypeRegistry.INSTANCE,
+            "test*",
+            readFieldCapsResponse("empty_field_caps_response.json"),
+            EsqlSession::specificValidity,
+            IndexResolver.PRESERVE_PROPERTIES,
+            IndexResolver.INDEX_METADATA_FIELD
+        );
```

Comment:
There's no easy way to do this kind of testing and, also, keeping it light.
The logic in `preAnalyzeIndices` calls in logic from `IndexResolver` which, in turn is creating a `field_caps` request (and possibly other requests related to aliases and CCS) that goes to ES. The call here is a simplified one where the field_caps response is emulated. What if the field_caps response format changes? This test will not protect us from this.

There is a balance here: test very thoroughly and create IT tests that use the exact code from `preAnalyzeIndices` or isolate the needed logic and do unit tests. I looked into IT tests because of the field_caps output I needed, but I decided it's too much and used unit tests instead. Also, there is another problem with the IT tests: they do not actually look at how the field_caps response look like and how the Analyzer is behaving in that case; the only thing you could assert on is the outcome of a query. OR create more elaborate mocking tests that allows you to run a `field_caps` call as close to reality as possible and then stop the ESQL logic at the analyzer level and look how the analysis is performed. Imo, it's too much work and resources for not so much gain.